openfga/v1/openfga_service.proto

```diff
@@ -1175,6 +1241,69 @@ message CheckResponse {
   string resolution = 2;
 }
 
+message BatchCheckRequest {
+  string store_id = 1 [
+    json_name = "store_id",
+    (google.api.field_behavior) = REQUIRED,
+    (validate.rules).string = {pattern: "^[ABCDEFGHJKMNPQRSTVWXYZ0-9]{26}$"},
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {example: "\"01YCP46JKYM8FJCQ37NMBYHE5X\""}
+  ];
+
+  repeated BatchCheckItem checks = 2 [
+    (google.api.field_behavior) = REQUIRED,
+    (validate.rules).repeated = {min_items: 1},
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {min_items: 1}
+  ];
+
+  string authorization_model_id = 3 [
+    json_name = "authorization_model_id",
+    (validate.rules).string = {
+      pattern: "^[ABCDEFGHJKMNPQRSTVWXYZ0-9]{26}$"
+      ignore_empty: true
+    },
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {example: "\"01G5JAVJ41T49E9TT3SKVS7X1J\""}
+  ];
+
+  ConsistencyPreference consistency = 4 [(validate.rules).enum.defined_only = true];
+}
+
+message BatchCheckItem {
+  CheckRequestTupleKey tuple_key = 1 [(google.api.field_behavior) = REQUIRED];
```

Comment:
Just noticed the OpenAPI spec has this down as `tupleKey`, I think we'll need to declare the `json_name` attribute.
```suggestion
  CheckRequestTupleKey tuple_key = 1 [
    json_name = "tuple_key",
    (google.api.field_behavior) = REQUIRED
  ];
```